src/main/java/com/blamejared/compat/tconstruct/TConstructHelper.java

```diff
@@ -0,0 +1,105 @@
+import slimeknights.tconstruct.library.smeltery.AlloyRecipe;
+import slimeknights.tconstruct.library.smeltery.ICastingRecipe;
+import slimeknights.tconstruct.library.smeltery.MeltingRecipe;
+import slimeknights.tconstruct.plugin.jei.CastingRecipeCategory;
+import slimeknights.tconstruct.plugin.jei.CastingRecipeWrapper;
+
+import java.util.*;
+
+/**
+ * Created by Jared for 1.10.2.
+ * Adapted by Rinart73 on 7/24/2017 for 1.11.2
+ */
+public class TConstructHelper {
+    public static List<AlloyRecipe> alloys = null;
+    public static LinkedList<ICastingRecipe> basinCasting = null;
+    public static LinkedList<ICastingRecipe> tableCasting = null;
+    public static List<MeltingRecipe> smeltingList = null;
+    public static List<DryingRecipe> dryingList = null;
+
+    public static Map<FluidStack, Integer> fuelMap = new HashMap<>();
+    public static Map<String, IModifier> modifiers = new THashMap<>();
+    public static Map<ResourceLocation, FluidStack> entityMeltingRegistry;
+
+    //this should bring a 'bridge' between IEntityDefinition and ResourceLocation
+    public static Map<String, ResourceLocation> entityLocationsByClassPath = new HashMap<>();
+
+    //allows to add casting recipes to the JEI, it contains images for table and basin
+    public static CastingRecipeCategory castingRecipeCategory = null;
+
+    //allows to delete casting recipes from JEI
+    public static Map<ICastingRecipe, CastingRecipeWrapper> castingRecipeWrappers = new HashMap<>();
+
+
+    static {
+        try {
+            alloys = ReflectionHelper.getStaticObject(TinkerRegistry.class, "alloyRegistry");
+            smeltingList = ReflectionHelper.getStaticObject(TinkerRegistry.class, "meltingRegistry");
+            dryingList = ReflectionHelper.getStaticObject(TinkerRegistry.class, "dryingRegistry");
+            basinCasting = ReflectionHelper.getStaticObject(TinkerRegistry.class, "basinCastRegistry");
+            tableCasting = ReflectionHelper.getStaticObject(TinkerRegistry.class, "tableCastRegistry");
+            modifiers = ReflectionHelper.getStaticObject(TinkerRegistry.class, "modifiers");
+            entityMeltingRegistry = ReflectionHelper.getStaticObject(TinkerRegistry.class, "entityMeltingRegistry");
+            fuelMap = ReflectionHelper.getStaticObject(TinkerRegistry.class, "smelteryFuels");
+
+            //Get ResourceLocations for Entities class paths and names
+            for(ResourceLocation r : EntityList.getEntityNameList()) {
+                Class<? extends Entity> clazz = EntityList.getClass(r);
+                if(clazz == null) continue;
+                entityLocationsByClassPath.put(clazz.getName(), r);
+            }
+
+            //Get all CastingRecipeWrapper-s and get CastingRecipes for them
+            //It's a terrible and slow way(~4ms), but TConstruct leaves no way to get CastingRecipeWrapper from CastingRecipe
+            ArrayList<String> categoryName = new ArrayList<>();
```

Comment:
So neither modtweaker or crafttweaker actually depend on JEI, and while I know that it is unlikely not to have JEI in a pack, it is possible, in which case this code will fail